bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/chart/ChartServlet.java

```diff
@@ -233,7 +223,8 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res) throws Ser
         }
 
         // Read out the parameter period, begin and end and save them.
-        Duration period = periodParam == null ? DEFAULT_PERIOD : PERIODS.getOrDefault(periodParam, DEFAULT_PERIOD);
+        Period period = convertToPeriod(periodParam, Period.ZERO);
+        Duration duration = convertToDuration(periodParam, DEFAULT_DURATION);
```

Comment:
I think you can collapse all there conversion methods into one that returns `TemporalAmount`. Otherwise LGTM.

Reply:
Good comment, I can simplify all that using `TemporalAmount`.

Reply:
Done